src/histogram/histograms.rs

```diff
@@ -78,6 +78,20 @@ impl<A: Ord> Histogram<A> {
     pub fn grid(&self) -> &Grid<A> {
         &self.grid
     }
+
+    /// Returns the cumulative distribution function of a histogram. 
+    /// Equivalent to the numpy histogram function cumsum
+    pub fn cumulative_sum(&self) -> ArrayD<usize> {
+        let mut cdf = self.counts.clone();
+        for i in 0..self.ndim() {
+            for j in 1..cdf.shape()[i] { 
+                let temp = cdf.index_axis(Axis(i), j - 1).to_owned();
```

Comment:
It's a shame that we have to use `to_owned` here, due to the borrow checker, even though the two slices are not-overlapping :disappointed:

Reply:
Actually, this can be avoided using `split_at`!

Reply:
So I just started trying to write that but found myself with two matrices of different sizes and thinking I'd like to get the last element of one for a given axis and the first of another and it felt too complicated.. But I might just not be picturing it in the right way :confused:

Reply:
The simplest approach would be to avoid the issue by using `.lanes_mut()`; something like this [*untested*]:

```rust
let mut cdf = self.counts.clone();
for ax in 0..cdf.ndim() {
    for lane in cdf.lanes_mut(Axis(ax)) {
        for i in 1..lane.len() {
            lane[i] += lane[i-1];
        }
    }
}
```

Reply:
If that works @jturner314, it's surely cleaner. Unfortunately it kind of goes the way you described @xd009642 using `split_at` :thinking: You need to split at `j` and then grab the last lane of the first part and the first lane of the second part.

Reply:
I just tested it, and my example does work (with the addition of `mut` before `lane`). It does rely on the element type being `Copy`, though; for general `A` you'd run into the same issue.

Fwiw, regarding the inconvenience of a solution based on `.split_at()`, the next release of `ndarray` will have a [`multislice!`](https://github.com/rust-ndarray/ndarray/blob/master/src/slice.rs#L653-L739) macro that makes it possible to cleanly take multiple disjoint, mutable slices simultaneously. This PR makes me realize that we should add a `multislice_axis!` macro too.